### ISSUE TITLE
Validation failed in API: name is not unique when editing Load Balancer

### ISSUE THREAD

**Issue by rubenmch** [rancher/rancher#3033]
I'm trying to edit a load balancer in the just released 0.50 version and I get the following error when trying to edit a load balancer:

Validation failed in API: name is not unique

I tried changing the load balancer name but I get the same error.

```
{"id":"c0cb585b-3460-48df-a3cc-8794824234d3","type":"error","links":{},"actions":{},"status":422,"code":"NotUnique","message":null,"detail":null,"fieldName":"name"}
```

I was able to edit the LB in the previous version (0.47).

**Comment by rubenmch** [comment]
It also happens with other types of services. New services aren't affected. Only services that existed before upgrading to Rancher 0.50.

**Comment by deniseschannon** [comment]
Can you verify that your UI has been updated? After upgrading, the UI might be cached. The UI version should be v0.74.0 with v0.50.0. I don't think that would be the issue, but sometimes that is an issue with people in upgraded setups. 

I wasn't able to reproduce your issue going from v0.47.0 to v0.50.0. I created some services in v0.47.0 and edited them in v0.50.0 with no issues in saving them.

**Comment by alena1108** [comment]
@deniseschannon @rubenmch there is a regression bug in v0.50.0 when you can't create service with the name that matches the name of the service that is removed in Rancher: 

https://github.com/rancher/rancher/issues/3037

So if the service you edit, happen to have a name = name of any removed service, you would hit the issue

**Comment by rubenmch** [comment]
@deniseschannon The UI is v0.74.0. The services were originally created in an older version. I first installed rancher in June or July, I think it was v0.32.0. The services were originally created in that version, but I cloned them and deleted the original several times when the option to upgrade didn't exist in rancher.

@alena1108 I get the same error message even if I change the name to a random name while editing one of the services that can't be edited. If I clone the service and delete the old service I can assign the old service name to the new service and I can edit the new service without issues.

Let me know if there's anything else I can check, otherwise to solve this I'll clone all the services and delete the old ones.

**Comment by deniseschannon** [comment]
I have finally been able to reproduce in one use case, but not sure if this is your use case. 

Steps to Reproduce:

1. Create LB1.
2. Create LB2. Edit name to same as LB1. 
3. Try to edit either original LB1 or duplicate LB1. 

Result: Validation failed in API: name and am unable to save anything.

@rubenmch @Ilya-Kuchaev Did either of you do anything like this where you have multiple services with the same name? I don't beleive the above use case is what is causing your issues, so would like to still figure out your root cause.

**Comment by deniseschannon** [comment]
The use case that I believe @rubenmch @Ilya-Kuchaev hit would be outlined below.

Have an existing service with name `foo`. Delete the service. In the DB, you will have a service of name `foo` with a `Removed` state. Note: This could have occurred in any previous release of Rancher that you were using.

You have an active service with name `foo`. If you try to edit this service, the validation for name is occurring against active and removed services, so it is failing. 

We have made the fix to stop the validation for name against removed services.

If you upgrade to v0.50.1 (when released), you will now be able to edit these services that you were previously unable to edit. 

Verified with v0.50.1-rc1.
